tools/troubleshooting-sharding-ddl-locks.md

```diff
@@ -0,0 +1,152 @@
+
+If the owner goes offline when the owner has finished executing the DDL statement but other DM-workers have not skipped this DDL statement. For the solution, see [Condition two: a DM-worker restarts](#condition-two-a-dm-worker-restarts-or-is-unreachable-temporarily).
+
+### Manual solution
+
+1. Run `show-ddl-locks` to obtain the information of the sharding DDL lock that is currently pending synchronization. 
+
+2. Run the `unlock-ddl-lock` command to specify the information of the lock to be unlocked manually.
+
+    - If the owner of this lock is offline, you can configure the `--owner` parameter to specify another DM-worker as the new owner to execute the DDL statement.
+
+3. Run `show-ddl-locks` to check whether this lock has been successfully unlocked.
+
+### Impact
+
+After you have manually unlocked the lock, it still might exist that the lock cannot be automatically synchronized when the next sharding DDL event is received, because the offline DM-workers are included in the task configuration information.
+
+Therefore, after you have manually unlocked the DM-workers, you need to use `stop-task`/`start-task` and the updated task configuration that does not include offline DM-workers to restart the task.
+
+> **Note:** If the DM-workers that went offline become online again after you run `unlock-ddl-lock`, it means: These DM-workers will synchronize the unlocked DDL operation again. (Other DM-workers that were not offline have synchronized the DDL statement.) The DDL operation of these DM-workers will try to match the subsequent synchronized DDL statements of other DM-workers. A match error of synchronizing sharding DDL statements of different DM-workers might occur.
+
+## Condition two: a DM-worker restarts (or is unreachable temporarily)
+
+Currently, the DDL unlocking process is not atomic, during which the DM-master schedules multiple DM-workers to execute or skip the sharding DDL statement and updates the checkpoint. Therefore, it might exist that after the owner finishes executing the DDL statement, a non-owner restarts before other DM-workers skip this DDL statement. At this time, the lock information on the DM-master has been removed but the restarted DM-worker has failed to skip this DDL statement and update the checkpoint.
+
+After the DM-worker restarts and runs `start-task`, it retries to synchronize the sharding DDL statement. But as other DM-workers have finished synchronizing this DDL statement, the restarted DM-worker cannot synchronize or skip this DDL statement.
+
+### Manual solution
+
+1. Run `query-status` to check the information of the sharding DDL statement that the restarted DM-worker is currently blocking. 
+
+2. Run `break-ddl-lock` to specify the DM-worker that is to break the lock forcefully.
+    
+    - Configure `skip` to specify the sharding DDL statement to be skipped.
```

Comment:
```suggestion
    - Specify `skip` to skip the sharding DDL statement.
```
because we only need to specify `skip`, and no need to specify the statement